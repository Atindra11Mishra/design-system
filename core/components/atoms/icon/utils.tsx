import { FontVariationType } from './Icon';

const getOpticalSize = (size?: number) => {
  if (size && size <= 20) {
    return 20;
  } else if (size && size >= 48) {
    return 48;
  }

  return size;
};

const iconVariantMapper: Record<string, object> = {
  more_horiz: {
    weight: 600,
    opticalSize: 24,
  },
  delete: {
    fill: 0,
  },
  radio_button_checked: {
    fill: 0,
  },
  radio_button_unchecked: {
    fill: 0,
  },
  schedule: {
    fill: 0,
  },
  help_outline: {
    fill: 0,
  },
  check_circle_outline: {
    fill: 0,
  },
  highlight_of: {
    fill: 0,
  },
  credit_card: {
    fill: 0,
  },
  delete_outline: {
    fill: 0,
  },
  bookmark_border: {
    fill: 0,
  },
  work_outline: {
    fill: 0,
  },
  alarm: {
    fill: 0,
  },
  arrow_circle_up: {
    fill: 0,
  },
  copyright: {
    fill: 0,
  },
  query_builder: {
    fill: 0,
  },
  card_membership: {
    fill: 0,
  },
  arrow_circle_down: {
    fill: 0,
  },
  change_history: {
    fill: 0,
  },
  aspect_ratio: {
    fill: 0,
  },
  calendar_view_month: {
    fill: 0,
  },
  alarm_on: {
    fill: 0,
  },
  thumb_down_off_alt: {
    fill: 0,
  },
  important_devices: {
    fill: 0,
  },
  turned_in_not: {
    fill: 0,
  },
  add_circle_outline: {
    fill: 0,
  },
  speed: {
    fill: 0,
  },
  content_copy: {
    fill: 0,
  },
  timelapse: {
    fill: 0,
  },
};

export const getFontVariation = (iconName: string, variations?: FontVariationType, type?: string, size?: number) => {
  const defaultVariant = {
    fill: type === 'outlined' ? 0 : 1,
    weight: 400,
    grade: 0,
    opticalSize: getOpticalSize(size),
  };

  return { ...defaultVariant, ...iconVariantMapper[iconName], ...variations };
};
