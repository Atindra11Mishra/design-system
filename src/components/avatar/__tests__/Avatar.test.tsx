import * as React from 'react';
import { render } from '@testing-library/react';
import { Avatar, AvatarProps as Props, AvatarSize } from '@/index';
import { AccentAppearance } from '@/common.type';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const appearances: AccentAppearance[] = [
  'primary',
  'alert',
  'warning',
  'success',
  'accent1',
  'accent2',
  'accent3',
  'accent4',
];

const sizes: AvatarSize[] = ['regular', 'tiny'];

describe('Avatar component', () => {
  const mapper = {
    appearance: valueHelper(appearances, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = render(<Avatar {...attr}>JD</Avatar>);
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Avatar component', () => {
  const mapper = {
    size: valueHelper(sizes, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = render(<Avatar {...attr}>JD</Avatar>);
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Avatar component', () => {
  it('render with regular size when no prop is provided', () => {
    const { getByTestId } = render(<Avatar />);
    expect(getByTestId('DesignSystem-Avatar')).toHaveClass('Avatar--regular');
  });

  it('render disabled avatar when no prop provided', () => {
    const { getByTestId } = render(<Avatar />);
    expect(getByTestId('DesignSystem-Avatar')).toHaveClass('Avatar--disabled');
  });

  it('renders avatar primary when appearance primary is passed', () => {
    const { getByTestId } = render(<Avatar appearance="primary" text="JD" />);
    expect(getByTestId('DesignSystem-Avatar')).toHaveClass('Avatar--primary');
  });

  it('renders tiny size when tiny primary is passed', () => {
    const { getByTestId } = render(<Avatar size="tiny" text="JD" />);
    expect(getByTestId('DesignSystem-Avatar')).toHaveClass('Avatar--tiny');
  });

  it('renders tooltip when tooltip is passed', () => {
    const { getByTestId } = render(<Avatar text="JD" tooltip="test avatar" />);
    expect(getByTestId('DesignSystem-Avatar').getAttribute('title')).toEqual('test avatar');
  });

  it('renders text when text prop is passed', () => {
    const { getByTestId } = render(<Avatar text="JD" />);
    expect(getByTestId('DesignSystem-Avatar').textContent).toMatch('JD');
  });

  it('render image alt text initial when wrong image src is passed', () => {
    const altText = 'test image';
    const { getByTestId } = render(<Avatar image={{ src: 'wrongurl.png', altText: altText }} />);
    expect(getByTestId('DesignSystem-Avatar').innerHTML).toBe(altText.charAt(0));
  });

  it('render with default icon when image src, alt text is not passed', () => {
    const { getByTestId } = render(<Avatar image={{ src: '', altText: '' }} />);
    expect(getByTestId('DesignSystem-Avatar--Icon')).toBeInTheDocument();
  });

  it('renders given icon when icon prop is passed', () => {
    const { getByTestId } = render(<Avatar icon="person" />);
    expect(getByTestId('DesignSystem-Avatar--Icon')).toBeInTheDocument();
  });

  it('renders fallback icon when there is no children', () => {
    const { getByTestId } = render(<Avatar />);
    expect(getByTestId('DesignSystem-Avatar--Icon')).toBeInTheDocument();
  });
});
