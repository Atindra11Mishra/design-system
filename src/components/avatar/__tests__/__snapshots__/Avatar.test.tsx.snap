// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component 
  appearance: "accent1"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Avatar Avatar--regular Avatar--accent1 Avatar--disabled overflow-hidden"
        data-test="DesignSystem-Avatar"
      >
        <i
          class="material-icons material-icons-round Icon Icon--white Avatar-content--accent1"
          data-test="DesignSystem-Avatar--Icon"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          person_round
        </i>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Avatar Avatar--regular Avatar--accent1 Avatar--disabled overflow-hidden"
      data-test="DesignSystem-Avatar"
    >
      <i
        class="material-icons material-icons-round Icon Icon--white Avatar-content--accent1"
        data-test="DesignSystem-Avatar--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        person_round
      </i>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component 
  appearance: "accent2"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Avatar Avatar--regular Avatar--accent2 Avatar--disabled overflow-hidden"
        data-test="DesignSystem-Avatar"
      >
        <i
          class="material-icons material-icons-round Icon Icon--white Avatar-content--accent2"
          data-test="DesignSystem-Avatar--Icon"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          person_round
        </i>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Avatar Avatar--regular Avatar--accent2 Avatar--disabled overflow-hidden"
      data-test="DesignSystem-Avatar"
    >
      <i
        class="material-icons material-icons-round Icon Icon--white Avatar-content--accent2"
        data-test="DesignSystem-Avatar--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        person_round
      </i>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component 
  appearance: "accent3"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Avatar Avatar--regular Avatar--accent3 Avatar--disabled overflow-hidden"
        data-test="DesignSystem-Avatar"
      >
        <i
          class="material-icons material-icons-round Icon Icon--white Avatar-content--accent3"
          data-test="DesignSystem-Avatar--Icon"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          person_round
        </i>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Avatar Avatar--regular Avatar--accent3 Avatar--disabled overflow-hidden"
      data-test="DesignSystem-Avatar"
    >
      <i
        class="material-icons material-icons-round Icon Icon--white Avatar-content--accent3"
        data-test="DesignSystem-Avatar--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        person_round
      </i>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component 
  appearance: "accent4"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Avatar Avatar--regular Avatar--accent4 Avatar--disabled overflow-hidden"
        data-test="DesignSystem-Avatar"
      >
        <i
          class="material-icons material-icons-round Icon Icon--white Avatar-content--accent4"
          data-test="DesignSystem-Avatar--Icon"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          person_round
        </i>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Avatar Avatar--regular Avatar--accent4 Avatar--disabled overflow-hidden"
      data-test="DesignSystem-Avatar"
    >
      <i
        class="material-icons material-icons-round Icon Icon--white Avatar-content--accent4"
        data-test="DesignSystem-Avatar--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        person_round
      </i>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component 
  appearance: "alert"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Avatar Avatar--regular Avatar--alert Avatar--disabled overflow-hidden"
        data-test="DesignSystem-Avatar"
      >
        <i
          class="material-icons material-icons-round Icon Icon--white Avatar-content--alert"
          data-test="DesignSystem-Avatar--Icon"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          person_round
        </i>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Avatar Avatar--regular Avatar--alert Avatar--disabled overflow-hidden"
      data-test="DesignSystem-Avatar"
    >
      <i
        class="material-icons material-icons-round Icon Icon--white Avatar-content--alert"
        data-test="DesignSystem-Avatar--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        person_round
      </i>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component 
  appearance: "primary"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Avatar Avatar--regular Avatar--primary Avatar--disabled overflow-hidden"
        data-test="DesignSystem-Avatar"
      >
        <i
          class="material-icons material-icons-round Icon Icon--white Avatar-content--primary"
          data-test="DesignSystem-Avatar--Icon"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          person_round
        </i>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Avatar Avatar--regular Avatar--primary Avatar--disabled overflow-hidden"
      data-test="DesignSystem-Avatar"
    >
      <i
        class="material-icons material-icons-round Icon Icon--white Avatar-content--primary"
        data-test="DesignSystem-Avatar--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        person_round
      </i>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component 
  appearance: "success"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Avatar Avatar--regular Avatar--success Avatar--disabled overflow-hidden"
        data-test="DesignSystem-Avatar"
      >
        <i
          class="material-icons material-icons-round Icon Icon--white Avatar-content--success"
          data-test="DesignSystem-Avatar--Icon"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          person_round
        </i>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Avatar Avatar--regular Avatar--success Avatar--disabled overflow-hidden"
      data-test="DesignSystem-Avatar"
    >
      <i
        class="material-icons material-icons-round Icon Icon--white Avatar-content--success"
        data-test="DesignSystem-Avatar--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        person_round
      </i>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component 
  appearance: "warning"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Avatar Avatar--regular Avatar--warning Avatar--disabled overflow-hidden"
        data-test="DesignSystem-Avatar"
      >
        <i
          class="material-icons material-icons-round Icon Icon--white Avatar-content--warning"
          data-test="DesignSystem-Avatar--Icon"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          person_round
        </i>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Avatar Avatar--regular Avatar--warning Avatar--disabled overflow-hidden"
      data-test="DesignSystem-Avatar"
    >
      <i
        class="material-icons material-icons-round Icon Icon--white Avatar-content--warning"
        data-test="DesignSystem-Avatar--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        person_round
      </i>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component 
  size: "regular"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Avatar Avatar--regular Avatar--secondary Avatar--disabled overflow-hidden"
        data-test="DesignSystem-Avatar"
      >
        <i
          class="material-icons material-icons-round Icon Icon--white Avatar-content--secondary"
          data-test="DesignSystem-Avatar--Icon"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          person_round
        </i>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Avatar Avatar--regular Avatar--secondary Avatar--disabled overflow-hidden"
      data-test="DesignSystem-Avatar"
    >
      <i
        class="material-icons material-icons-round Icon Icon--white Avatar-content--secondary"
        data-test="DesignSystem-Avatar--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        person_round
      </i>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar component 
  size: "tiny"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Avatar Avatar--tiny Avatar--secondary Avatar--disabled overflow-hidden"
        data-test="DesignSystem-Avatar"
      >
        <i
          class="material-icons material-icons-round Icon Icon--white Avatar-content--secondary"
          data-test="DesignSystem-Avatar--Icon"
          role="button"
          style="font-size: 12px; width: 12px;"
        >
          person_round
        </i>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Avatar Avatar--tiny Avatar--secondary Avatar--disabled overflow-hidden"
      data-test="DesignSystem-Avatar"
    >
      <i
        class="material-icons material-icons-round Icon Icon--white Avatar-content--secondary"
        data-test="DesignSystem-Avatar--Icon"
        role="button"
        style="font-size: 12px; width: 12px;"
      >
        person_round
      </i>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
